installer/scripts/service_control

```diff
@@ -283,12 +471,16 @@ enable_omsagent_service()
             elif [ -x /sbin/chkconfig ]; then
                 /sbin/chkconfig --add $OMSAGENT_WS > /dev/null
             else
-                echo "Could not find a service controller to configure the OMS Agent Service."
-                exit 1
+                echo "FATAL:  Could not find a service controller to configure the OMS Agent Service."
+                exit 9
             fi
         fi
-        touch  $CONF_DIR/.service_registered
+        WS_STATUS=$?
+        if [ "$WS_STATUS" -eq 0 ]; then
```

Comment:
integer comparison: `$WS_STATUS -eq 0`

Reply:
Yes, I think what I've heard is unless you are always sure, like with $?, it is better to do "$WS_STATUS" -eq "0" or equivalent.  I'm trying to stick to that as long as it's POSIX and not Bash.  In Bash, which I don't think we should follow here, [[]] constructs are recommended, which are much nicer.

Reply:
I should have posted my commend about the remove_* procedure in here somewhere.  Please see that Laura and I found the original state worked better because remove_PID* procedure succeeds both at times when omsagent is running, and at times when it is not, by design.  Comments have been added.

Reply:
This is still comparing a string to an integer - have you fixed this and I'm looking at the wrong commit?

Reply:
This is still comparing a string to an integer - please fix this!

Reply:
Going through the entire file...Okay, I took all those kinds out (anything with -eq 0, -ne 0, -gt, -lt -ge -le 0, EXCEPT those where we know they are numeric like `id -u`, $COUNTER, and $?).